### PR TITLE
Add Documentation / Maintenance subgroups to release notes

### PR DESCRIPTION
**Pull request by gasman**
Reorganise the 4.2 release notes and changelog to include new sections for documentation and maintenance. This helps to highlight the most relevant changes from the perspective of a developer wanting to know what they can expect from an upgrade, while still recognising less visible contributions. Also update the "committing code" docs to explain the new classification scheme.

### CODE REVIEW

File: CHANGELOG.txt [hunk 1]
```diff
@@ -43,6 +24,25 @@ Changelog
...
+ * Maintenance: Clean up duplicate JavaScript for the `escapeHtml` function (Jordan Rob)
+ * Maintenance: Clean up Prettier & Eslint usage for search promotions formset JS file (LB (Ben Johnston))
+ * Maintenance: Ensure that translation file generation ignores JavaScript unit tests and clean up unit tests for Django gettext utils (LB (Ben Johnston))
+ * Maintenance: Migrated `initButtonSelects` from core.js to own TypesScript file and add unit tests (Loveth Omokaro)
+ * Maintenance: Migrated `initSkipLink` util to TypeScript and add JSDoc & unit tests (Juliet Adeboye)
+ * Maintenance: Clean up some unused utility classes and migrate `unlist` to Tailwind utility class `w-list-none` (Loveth Omokaro)
+ * Maintenance: Clean up linting on legacy code and add shared util `hasOwn` in TypeScript (Loveth Omokaro)
+ * Maintenance: Remove unnecessary box-sizing: border-box declarations in SCSS (Albina Starykova)
+ * Maintenance: Migrated `initTooltips` to TypeScript add JSDoc and unit tests (Fatuma Abdullahi)
+ * Maintenance: Migrated `initTagField` from core.js to own TypeScript file and add unit tests (Chisom Okeoma)
```
Comment (lb-) [inline CHANGELOG.txt:45]: Nitpicky but should we call this prefix `Chore:` ?

Comment (lb-) [reply CHANGELOG.txt:45]: Ignore this. Maintenance makes more sense after reading the rest of.the PR.

**Review by lb- — APPROVED** [review]
Reads really well - thanks and I am all for this change.

**Comment by lb-** [comment]
Now the difficult part - is this a maintenance or docs change 😅

**Comment by gasman** [comment]
Merged in 44933d06c47dfebfa0152982027dd148128dc87f

**Pull request closed without merging**